src/control/InstantaneousTrajectoryExecutor.cpp

```diff
@@ -23,7 +23,8 @@ InstantaneousTrajectoryExecutor::~InstantaneousTrajectoryExecutor()
 }
 
 //==============================================================================
-void InstantaneousTrajectoryExecutor::validate(trajectory::TrajectoryPtr traj)
+void InstantaneousTrajectoryExecutor::validate(
+    const trajectory::ConstTrajectoryPtr& traj)
```

Comment:
If we throw an exception for nullptr `traj`, there is no point to take `traj` as a pointer. Also, this function doesn't take the ownership so taking it as `shared_ptr` is unnecessary. Taking this as const reference would make more sense because (1) it can enforce to pass not a `nullptr` here, and (2) it allows to pass a raw pointer (a trajectory doesn't need to be shared_ptr to call this function).

One possibility we still want this as `shared_ptr` would be the case that we foresee a derived class of `TrajectoryExecutor` would want to take the ownership to validate a trajectory, which I don't think so. Let me know if I'm missing something here.

If we change `traj` to const reference, please remove `nullptr` check.